Sources/Plasma/FeatureLib/pfConsoleCore/pfConsoleCmd.cpp

```diff
@@ -367,118 +328,72 @@ int  pfConsoleCmdGroup::IterateCommands(pfConsoleCmdIterator* t, int depth)
+pfConsoleCmd::pfConsoleCmd(const ST::string& group, ST::string name,
+                            const ST::string& paramList, ST::string help, 
+                            pfConsoleCmdPtr func)
+    : fName(std::move(name)), fHelpString(std::move(help)), fFunction(func),
+        fNext(nullptr), fPrevPtr(nullptr), fParentGroup(nullptr)
 {
-    fNext = nullptr;
-    fPrevPtr = nullptr;
-
-    fFunction = func;
-    fLocalOnly = localOnly;
-    
-    hsStrncpy( fName, name, sizeof( fName ) );
-    fHelpString = help;
-
     ICreateSignature( paramList );
-    Register( group, name );
+    Register(group);
 }
 
 pfConsoleCmd::~pfConsoleCmd()
 {
-    int     i;
-
-
-    for( i = 0; i < fSigLabels.GetCount(); i++ )
-    {
-        delete [] fSigLabels[ i ];
-    }
     Unregister();
-    
-    fSignature.Reset();
-    fSigLabels.Reset();
 }
 
 //// ICreateSignature ////////////////////////////////////////////////////////
 //  Creates the signature and sig labels based on the given string.
 
-void    pfConsoleCmd::ICreateSignature(const char *paramList )
+void pfConsoleCmd::ICreateSignature(const ST::string& paramList)
 {
-    static char seps[] = " :-";
-
-    char    params[ 256 ];
-    char    *ptr, *nextPtr, *tok, *tok2;
-    int     i;
-
+    fSignature.clear();
+    fSigLabels.clear();
 
-    /// Simple check
-    if (paramList == nullptr)
-    {
-        fSignature.Push( kAny );
-        fSigLabels.Push(nullptr);
+    if (paramList.empty()) {
         return;
     }
 
-    /// So we can do stuff to it
-    hsAssert( strlen( paramList ) < sizeof( params ), "Make the (#*$& params string larger!" );
-    hsStrcpy( params, paramList );
-
-    fSignature.Reset();
-    fSigLabels.Reset();
-
     /// Loop through all the types given in the list
-    ptr = params;
-    do
-    {
-        /// Find break
-        nextPtr = strchr( ptr, ',' );
-        if (nextPtr != nullptr)
-        {
-            *nextPtr = 0;
-            nextPtr++;
-        }
+    for (const auto& param : paramList.split(',')) {
+        std::vector<ST::string> parts = param.tokenize(" :-");
+        hsAssert(!parts.empty(), "Bad parameter list for console command!");
+        hsAssert(parts.size() <= 2, "Bad parameter list for console command!");
 
-        /// Do this param
-        tok = strtok( ptr, seps );
-        if (tok == nullptr && ptr == params)
-            break;
-
-        hsAssert(tok != nullptr, "Bad parameter list for console command!");
-        tok2 = strtok(nullptr, seps);
-
-        if (tok2 != nullptr)
-        {
+        if (parts.size() == 2) {
             // Type and label: assume label second
-            fSigLabels.Push( hsStrcpy( tok2 ) );            
+            fSigLabels.emplace_back(std::move(parts[1]));
+        } else {
+            fSigLabels.emplace_back();
         }
-        else
-            fSigLabels.Push(nullptr);
 
         // Find type
+        ST::string type = std::move(parts[0]);
```

Comment:
```suggestion
        ST::string& type = parts[0];
```

This seems to just be for self documentation, so no need to actually move the contents of the string.